app/src/main/java/com/example/atheneum/activities/ViewEditBookPhotos.java

```diff
@@ -0,0 +1,157 @@
+ */
+public class ViewEditBookPhotos extends AppCompatActivity {
+    private static final String TAG = ViewEditBookPhotos.class.getSimpleName();
+    public static final String INTENT_BOOK_ID =
+            ViewEditBookPhotos.class.getCanonicalName() + ":BOOK_ID";
+    public static final String INTENT_OWNER_USER_ID =
+            ViewEditBookPhotos.class.getCanonicalName() + ":OWNER_USER_ID";
+
+    private CameraHandler cameraHandler;
+    private BookPhotosViewModel bookPhotosViewModel;
+    private FloatingActionButton fab;
+    private BookPhotosListAdapter bookPhotosListAdapter;
+    private RecyclerView bookPhotosGalleryRecyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_edit_book_photos);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        final String bookID = getIntent().getStringExtra(INTENT_BOOK_ID);
+        final String ownerUserID = getIntent().getStringExtra(INTENT_OWNER_USER_ID);
+
+        if (bookID == null || bookID.equals("")) {
+            Log.w(TAG, "bookID is not valid! This should never happen!");
+            return;
+        } else if (ownerUserID == null || ownerUserID.equals("")) {
+            Log.w(TAG, "ownerUserID is not valid! This should never happen!");
+            return;
+        } else if (!FirebaseAuthUtils.isCurrentUserAuthenticated()) {
+            Log.w(TAG, "user is not authenticated! This should not happen!");
+            return;
+        }
+
+        bookPhotosListAdapter = new BookPhotosListAdapter();
+        bookPhotosGalleryRecyclerView = findViewById(R.id.book_photos_gallery);
+        GridLayoutManager layoutManager = new GridLayoutManager(getApplicationContext(), 3);
+        ItemOffsetDecoration itemDecoration = new ItemOffsetDecoration(getApplicationContext(), R.dimen.item_offset);
+        bookPhotosGalleryRecyclerView.addItemDecoration(itemDecoration);
+        bookPhotosGalleryRecyclerView.setAdapter(bookPhotosListAdapter);
+        bookPhotosGalleryRecyclerView.setLayoutManager(layoutManager);
+
+        Log.i(TAG, "bookID: " + bookID);
+        BookPhotosViewModelFactory factory = new BookPhotosViewModelFactory(bookID);
+        bookPhotosViewModel = ViewModelProviders.of(this, factory).get(BookPhotosViewModel.class);
+        bookPhotosViewModel.getPhotosLiveData().observe(this, new Observer<ArrayList<Photo>>() {
+            @Override
+            public void onChanged(@Nullable ArrayList<Photo> photos) {
+                bookPhotosListAdapter.submitList(photos);
+            }
+        });
+
+        if (FirebaseAuthUtils.getCurrentUser().getUid().equals(ownerUserID)) {
+            cameraHandler = CameraHandler.newInstance(this);
+
+            bookPhotosListAdapter.setOnItemClickListener(new BookPhotosListAdapter.OnItemClickListener() {

```

Comment:
Any particular reason  you decided to have click on the photo open the camera? Just kind of curious because i thought it was a bit strange.

Reply:
Clicking on the photo lets you edit it. When you take the picture, the photo updates in the Recyclerview. When you cancel, nothing happens. That's probably not the best UI since it's really confusing.

I could probably change it so that there is a dialog box that lets you edit/delete the photo. 

It might be better to even have a separate activity for viewing/editing/deleting a particular photo for a book. But that might depend on how good having a scaled up image of the book looks like since the images are really low res.

Reply:
Fixed. This was a really bad UI design. So I just changed it so that clicking a photo takes you to a dialog box that would ask you to edit the photo while a long click takes you to a dialog box asking you to delete a photo.